src/device-registry/models/Device.js

```
@@ -237,161 +237,167 @@
-        excludeSimilarCharacters: true,
-      });
-      updateData.access_code = access_code.toUpperCase();
-    }
+      // Handle $addToSet for device_codes, previous_sites, and pictures
+      const addToSetUpdates = {};
 
-    // Handle $addToSet for device_codes, previous_sites, and pictures
-    const addToSetUpdates = {};
+      if (updateData.device_codes) {
+        addToSetUpdates.device_codes = { $each: updateData.device_codes };
+        delete updateData.device_codes; // Remove from main update object
```

Biome (1.9.4)

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)